src/main/groovy/io/seqera/wave/service/blob/impl/KubeTransferStrategy.groovy

```diff
@@ -46,19 +54,50 @@ class KubeTransferStrategy implements TransferStrategy {
     @Inject
     private K8sService k8sService
 
+    @Inject
+    private CleanupStrategy cleanup
+
+    @Inject
+    @Named(TaskExecutors.IO)
+    private ExecutorService executor
+
     @Override
     BlobCacheInfo transfer(BlobCacheInfo info, List<String> command) {
-        final podName = podName(info)
-        final pod = k8sService.transferContainer(podName, blobConfig.s5Image, command, blobConfig)
-        final terminated = k8sService.waitPod(pod, blobConfig.transferTimeout.toMillis())
-        final stdout = k8sService.logsPod(podName)
-        return terminated
+        final name = getName(info)
+        final job = k8sService.transferJob(name, blobConfig.s5Image, command, blobConfig)
+        final podList = k8sService.waitJob(job, blobConfig.transferTimeout.toMillis())
+        final size = podList.items.size()
+
+        if( size < 1 )
+            throw new TransferTimeoutException("Transfer job timed out")
+
+        // Find the latest created pod among the pods associated with the job
+        def latestPod = podList.getItems().get(0)
+        for (def pod : podList.items) {
+            if (pod.metadata.creationTimestamp.isAfter(latestPod.metadata.creationTimestamp)) {
+                latestPod = pod
+            }
+        }
+
+        final podName = latestPod.metadata.name
+        final pod = k8sService.getPod(podName)
+        final terminated = k8sService.waitPod(pod, name, blobConfig.transferTimeout.toMillis())
```

Comment:
Why is it needed to wait for the pod, if there's a wait for the job above?

Reply:
This is here to use the existing logic, pod is already completed so it won't be a wait, just to get the terminated object from this function

Reply:
Ok, just added `findLatestPod` to make it more readable